sources/osgGA/FirstPersonManipulator.js

```diff
@@ -56,14 +57,17 @@ FirstPersonManipulator.prototype = MACROUTILS.objectInherit( Manipulator.prototy
         this._zoom = new OrbitManipulator.Interpolator( 1 );
 
         this._stepFactor = 1.0; // meaning radius*stepFactor to move
-        this._target = Vec3.create();
         this._angleVertical = 0.0;
         this._angleHorizontal = 0.0;
 
         // tmp value use for computation
         this._tmpGetTargetDir = Vec3.create();
 
-        this._rotBase = Matrix.create();
+        // vr controls
+        this._vrEnable = false;
```

Comment:
The only way to have several controllers controlling a manipulator is:
- the manipulator only exposes a view matrix and it's the controllers responsability to handle it
- the manipulator exposes custom stuffs (like angleVertical/angleHorizontal for example), in that case all the controls should edit those

The second solution is limited because you can't do rool axis vr movement with only 2 angles.
The first solution is better imho, but it raises the question of the limit/constraints, it's not clear if it's the manipulator or controllers  reponsability to handle these.

I'm thinking it should be the controllers, for example in fps manipulator you don't want any constraints when you move your head in VR, but with the mouse controller you never want to go on the other side.

Also with the first solution, I'm not sure of the benefit of having several manipulator if they only exposes a viewMatrix to the controllers (in that case it will be the controllers that will do the heavy works).